File: backend/app/schemas/notification.py
from pydantic import BaseModel
from typing import Optional, List

class NotificationItem(BaseModel):
    id: str
    type: str  # "VERIFICATION_REQUIRED" | "RESTRICTED_ACCESS" | "INTEGRITY_MISMATCH" | "CASE_UPDATE" | "VERIFICATION_SUCCESS" | "CASE_ASSIGNMENT"
    title: str
    message: str
    case_id: Optional[str] = None
    case_number: Optional[str] = None
    document_id: Optional[str] = None
    timestamp: str
    severity: str  # "INFO" | "WARNING" | "CRITICAL" | "SUCCESS"
    target_url: str
    read: bool = False

class NotificationsResponse(BaseModel):
    total: int
    unread_count: int
    notifications: List[NotificationItem]

File: Frontend/src/types/notification.ts
export type NotificationType =
  | 'VERIFICATION_REQUIRED'
  | 'RESTRICTED_ACCESS'
  | 'INTEGRITY_MISMATCH'
  | 'CASE_UPDATE'
  | 'VERIFICATION_SUCCESS'
  | 'CASE_ASSIGNMENT';

export type NotificationSeverity = 'INFO' | 'WARNING' | 'CRITICAL' | 'SUCCESS';

export interface NotificationItem {
  id: string;
  type: NotificationType;
  title: string;
  message: string;
  caseId?: string;
  caseNumber?: string;
  documentId?: string;
  timestamp: string;
  severity: NotificationSeverity;
  targetUrl: string;
  read?: boolean;
}

export interface NotificationsResponse {
  total: number;
  unreadCount: number;
  notifications: NotificationItem[];
}
